wait_all.go

```diff
@@ -0,0 +1,37 @@
+package asynctask
+
+// WaitAll block current thread til all task finished.
+// it return immediately after receive first error.
+func WaitAll(tasks ...*TaskStatus) error {
+	errorCh := make(chan error)
+	errorChClosed := false
+	defer func() {
+		errorChClosed = true
+		close(errorCh)
+	}()
+
+	runningTasks := 0
+	for _, tsk := range tasks {
+
+		go func(tsk *TaskStatus) {
+			_, err := tsk.Wait()
+			if !errorChClosed {
```

Comment:
Can the `defer`'d func that closes the channel ever be hit before all tasks have completed?